crates/sui-sdk/src/error.rs

```diff
@@ -12,13 +13,13 @@ pub enum Error {
     #[error(transparent)]
     RpcError(#[from] jsonrpsee::core::Error),
     #[error(transparent)]
+    JsonRpcError(JsonRpcError),
+    #[error(transparent)]
     BcsSerialisationError(#[from] bcs::Error),
     #[error(transparent)]
     UserInputError(#[from] UserInputError),
     #[error("Subscription error : {0}")]
     Subscription(String),
-    #[error("Encountered error when confirming tx status for {0:?}, err: {1:?}")]
-    TransactionConfirmationError(TransactionDigest, jsonrpsee::core::Error),
```

Comment:
Is this removal intended for this PR? It doesn't seem to fit with the rest of the change.

Reply:
Ah I now see the connection with this PR of removing this transaction, but two questions:

- What about `RpcError`?
- Are we losing structure/information by replacing `TransactionConfirmationError` (a kind of JSON RPC error) with a more generic message?

Reply:
I'm removing TransactionConfirmationError because it doesn't show up in use anywhere. 

RpcError will still exist. I think eventually we'll want to retire this, especially with the Graphql server coming up, so RpcError -> JsonRpcError, GraphqlError

Reply:
Thanks for clarifying.  FWIW, I wouldn't expect there to be a type that has both `JsonRpcError` and `GraphqlError` -- I expect we may want to take a second go at `sui-sdk` once the GraphQL Server has stabilised instead, so we can re-architect it to take advantage of the new API.